docs/library/userguide.md

```diff
@@ -95,8 +94,24 @@ qref-render my_program.yaml my_program_graph.svg
 The `qref-render` tool supports `yaml` and `json` input formats, and all
 output formats supported by [graphviz](https://graphviz.org/).
 
-If, instead of using CLI, you'd like to invoke QREF's rendering capabilities
-from Python script, you can look at [qref.experimental.rendering][qref.experimental.rendering]
-module which exposes experimental API for performing the same task as `qref-render`.
+If you prefer to use QREF's rendering capabilities from a Python script instead of the CLI, you can use the [`qref.experimental.rendering`](qref.experimental.rendering) module,  which performs the same task as `qref-render`. Here, we demonstrate how to use the rendering module to visualize quantum circuits for preparing arbitrary quantum states in alias sampling. To learn more about the algorithm, please refer to the tutorial for [Bartiq](https://psiq.github.io/bartiq/latest/tutorials/02_alias_sampling_basic/).
+
+We will use the `yaml` file `alias_sampling_detailed_fixed_resources.yaml` as input to generate a graph of the circuit:
 
+```python
+import yaml
+from qref import SchemaV1
+from qref.experimental.rendering import to_graphviz
 
+# Load the YAML file
+with open("../examples/alias_sampling_detailed_fixed_resources.yaml", "r") as f:
+    data = yaml.safe_load(f)
+
+# Validate the schema and convert to Graphviz object
+program = SchemaV1.model_validate(data)
+gv_object = to_graphviz(program)
+
+# Render the Graphviz object to a PNG file
+gv_object.render("as",format="png")
```

Comment:
[nitpick] I'd change the name from `as` to `alias_sampling` – it might not be immediately obvious to someone what `as` stands for.